### PR TITLE
Don't expect istio-system to have an injection label

### PR DESCRIPTION
**Pull request by esnible**
~~Resolves https://github.com/istio/istio/issues/30874~~
Starts work on https://github.com/istio/istio/issues/30874 by skipping _istio-system_ when checking for injection labels.

Because _istio-system_ does not have an injection label, yet also doesn't run user workloads, don't expect it to have an `istio-injection` label.

### CODE REVIEW

File: galley/pkg/config/analysis/analyzers/util/config.go [hunk 1]
```diff
@@ -32,6 +33,8 @@ var (
 		constants.KubePublicNamespace,
 		constants.KubeNodeLeaseNamespace,
 		constants.LocalPathStorageNamespace,
+
```
Comment (howardjohn) [inline galley/pkg/config/analysis/analyzers/util/config.go:36]: this list is used for a lot of other purposes, seems like this may negatively impact those

Comment (esnible) [reply galley/pkg/config/analysis/analyzers/util/config.go]: I missed that the list is used by _bugreport.go_.  I feel like we analysis needs a function to tell if a namespace doesn't run workloads and this is the function.  I made a slight change so it doesn't affect _bugreport.go_.  If you had something else in mind please elaborate.

Comment (howardjohn) [reply galley/pkg/config/analysis/analyzers/util/config.go]: I thought it was more original, but looks like it was just bugreport

File: galley/pkg/config/analysis/analyzers/util/config.go [hunk 2]
```diff
@@ -37,7 +38,7 @@ var (
 
 // IsSystemNamespace returns true for system namespaces
 func IsSystemNamespace(ns resource.Namespace) bool {
```
Comment (howardjohn) [inline galley/pkg/config/analysis/analyzers/util/config.go:40]: IsSystemNamespace is used by bug report as well

Comment (esnible) [reply galley/pkg/config/analysis/analyzers/util/config.go:40]: @howardjohn Yes, but it is only used to skip running analysis!  This change will just make it skip _istio-system_, which shouldn't be analyzed anyway.

**Comment by howardjohn** [comment]
Why not analyze istio-system? there are a lot of configs that should be
analyzed there

**Review by howardjohn — APPROVED** [review]

**Review by shamsher31 — APPROVED** [review]

**Pull request merged**